internalsdk/session_model.go

```diff
@@ -1453,44 +1530,43 @@ func reportIssue(session *SessionModel, email string, issue string, description
 func checkAdsEnabled(session *SessionModel) error {
 	log.Debugf("Check ads enabled")
 	// Check if ads is enable or not
-	adsEnable, err := pathdb.Get[bool](session.db, pathAdsEnabled)
+	isPro, err := session.IsProUser()
 	if err != nil {
-		return log.Errorf("Error while getting ads enabled %v", err)
+		return err
 	}
-	if !adsEnable {
+	if isPro {
 		return pathdb.Mutate(session.db, func(tx pathdb.TX) error {
-			return pathdb.Put(tx, pathShouldShowGoogleAds, false, "")
+			return pathdb.Put[string](tx, pathShowAds, "", "")
 		})
 	}
-
-	// if enable
+	// if user is not pro check if user provdied all permission
 	hasAllPermisson, err := pathdb.Get[bool](session.db, pathHasAllNetworkPermssion)
 	if err != nil {
 		return err
 	}
 	// If the user doesn't have all permissions, disable Google ads:
 	if !hasAllPermisson {
 		log.Debugf("User has not given all permission")
-
 		return pathdb.Mutate(session.db, func(tx pathdb.TX) error {
-			return pathdb.Put(tx, pathShouldShowGoogleAds, false, "")
+			return pathdb.Put[string](tx, pathShowAds, "", "")
 		})
 	}
-	log.Debugf("User has given all permission")
-	isPro, err := session.IsProUser()
-	if err != nil {
-		return err
+	// If the user has all permissions but is not a pro user, enable ads:
+	googleAdsEnable, _ := pathdb.Get[bool](session.db, pathShouldShowGoogleAds)
+	tapSellAdsEnable, _ := pathdb.Get[bool](session.db, pathTapSellAdsEnabled)
```

Comment:
Should we check for errors here or are the path guaranteed to have values at this point?

Reply:
It's better to check for errors always, added error check.